packages/plugins/lib/plugin.ts

```diff
@@ -0,0 +1,168 @@
+const TruffleError: any = require("@truffle/error");
+const originalRequire = require("original-require");
+import path from "path";
+
+// plugin configuration as specified in projects' truffle-config.js
+// supports either plain module name as string, or wrapper object
+export type RawPluginConfig = PluginConfig | string;
+
+// internal representation of a particular plugin configuration
+export interface PluginConfig {
+  module: string;
+}
+
+/**
+ * Returns true or false based on whether or not a particular plugin
+ * resolves successfully
+ */
+export const resolves = (module: string): boolean => {
+  try {
+    originalRequire.resolve(module);
+    return true;
+  } catch (_) {
+    return false;
+  }
+};
+
+/**
+ * Takes a list of raw plugin configurations and returns a list of normalized
+ * internal representations
+ */
+export const normalizeConfigPlugins = (
+  plugins: RawPluginConfig[]
+): PluginConfig[] => {
+  const map: Map<string, PluginConfig> = new Map([]);
+
+  const normalized = plugins.map((plugin: RawPluginConfig) =>
+    typeof plugin === "string" ? { module: plugin } : plugin
+  );
+
+  for (const plugin of normalized) {
+    // fatal error if we can't load a plugin listed in truffle-config.js
+    if (!resolves(plugin.module)) {
+      throw new TruffleError(
+        `\nError: ${module} listed as a plugin, but not found in global or local node modules!\n`
```

Comment:
Same about `\n` and `OS.EOL`, should we care about this? Does it matter? I forgot what the outcome of our discussion about this was.